### PR TITLE
Added in typeDefs to models in schema.js

### PR DESCRIPTION
**Pull request by PaBoba**

### CODE REVIEW

File: server/graphql/schema.js [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
+    ): Message
+    updateUser(username: String!, email: String!, password: String!): User
+    login(email: String!, password: String!): Auth
   }
 
-  // Define other types here
+  type User {
+    _id: ID
+    username: String
+    email: String
```
Comment (stevendreed) [inline server/graphql/schema.js]: we need to update the db to include emails, or remove this

File: server/graphql/schema.js [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
+  }
+
+  type Message {
+    _id: ID
+    sender: String
+    message_id: UUID
+    content: String
+    reaction: json
+    thread: UUID
+    location: UUID
```
Comment (stevendreed) [inline server/graphql/schema.js]: sorry for the confusion: we're storing as Strings and will use `js` logic to store UUIDs within them, + hooks to verify data stored are UUIDs

File: server/models/message.js [hunk 3]
```diff
@@ -1,34 +1,37 @@
...
+  {
+    sender: {
+      type: String,
+      ref: "user",
     }, // maps to the username which sent the message
-    message_id: { 
-        type: UUID,
-        unique: true,
+    message_id: {
+      type: UUID,
```
Comment (stevendreed) [inline server/models/message.js]: as before: we're storing our UUIDs as the `string` data type, not a UUID datatype

File: server/models/message.js [hunk 4]
```diff
@@ -1,34 +1,37 @@
...
-        trim: true, // remove leading & trailing spaces
+      type: String,
+      trim: true, // remove leading & trailing spaces
     },
     reaction: json, // key-val pairs of user:'some user', reaction:'👍'
-    thread: { 
-        type: UUID, 
-        ref: 'message' 
+    thread: {
+      type: UUID,
```
Comment (stevendreed) [inline server/models/message.js:17]: UUID confusion: a novel by Steven

File: server/models/message.js [hunk 5]
```diff
@@ -1,34 +1,37 @@
...
+      ref: "message",
     },
-    location: { 
-        type: UUID, 
-        ref: 'chatroom' 
+    location: {
+      type: UUID,
```
Comment (stevendreed) [inline server/models/message.js:21]: UUID

**Review by bakuwaku-cloud — DISMISSED** [review]
in the schema.sql 
- reference object ids are invalid. dont use uuid, change to subsequent [ObjectId] (i.e. chatroom,user,etc) or String
- not advisable to expose email or password for user. remove it from api. use of email & password in mutations is fine 
-remove message_id field in mutations
- define auth type in the mutations
- add chatroom mutation 
-add missing types: chatroom & tags
`type Chatroom {
    _id: ID
    title: String
    tags: [tag]
    icon: String
    activeUsers: [user]
    createdAt: String
  }

  type Tag {
    _id: ID
    name: String
  }`

**Review by stevendreed — COMMENTED** [review]
see files for comments

**Review by stevendreed — APPROVED** [review]
LGTM

**Pull request merged**